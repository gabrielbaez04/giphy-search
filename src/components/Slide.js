import React from 'react';
import PropTypes from 'prop-types';

const Slide = ({ image }) => {
    const styles = {
        backgroundImage: `url(${image})`,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: '50% 60%'
    }
    return <div className="slide" style={styles}></div>
}

Slide.propTypes = {
    image: PropTypes.string.isRequired
}

export default Slide;