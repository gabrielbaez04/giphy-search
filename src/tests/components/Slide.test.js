import { shallow } from 'enzyme';
import Slide from '../../components/Slide';
import React from 'react';

test('should render Slide correctly', () => {
    const wrapper = shallow(<Slide image="test" />);
    expect(wrapper).toMatchSnapshot();
});
