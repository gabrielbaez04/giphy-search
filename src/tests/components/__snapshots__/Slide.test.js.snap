// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Slide correctly 1`] = `
<div
  className="slide"
  style={
    Object {
      "backgroundImage": "url(test)",
      "backgroundPosition": "50% 60%",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
/>
`;
